gtsfm/data_association/point3d_initializer.py

```diff
@@ -55,6 +56,39 @@ class TriangulationParam(Enum):
     RANSAC_TOPK_BASELINES = 3  # deterministically choose hypotheses with largest estimate baseline
 
 
+class TriangulationOptions(NamedTuple):
+    """Options for triangulation RANSAC solver."""
+
+    reproj_error_threshold: float
+    min_track_len: int
+    mode: TriangulationParam
+
+    # RANSAC parameters
+    min_inlier_ratio: float = 0.5
+    confidence: float = 0.9999
+    dyn_num_hypotheses_multiplier: float = 3.0
+    max_num_hypotheses: int = sys.maxsize
+
+    def __check_ransac_params(self) -> None:
+        """Check that input value are valid"""
+        assert self.reproj_error_threshold > 0
+        assert 0 < self.min_inlier_ratio < 1
+        assert 0 < self.confidence < 1
+        assert self.dyn_num_hypotheses_multiplier > 0
+        assert self.max_num_hypotheses > 0
+
+    def num_ransac_hypotheses(self) -> int:
+        """Compute maximum number of hypotheses."""
+        self.__check_ransac_params()
+        dyn_num_hypotheses = int(
+            np.log(1 - self.confidence)
+            / np.log(1 - (1 - self.min_inlier_ratio) ** NUM_SAMPLES_PER_RANSAC_HYPOTHESIS)
+            * self.dyn_num_hypotheses_multiplier
+        )
+        return min(self.max_num_hypotheses, dyn_num_hypotheses)
+        # return self.max_num_hypotheses
```

Comment:
commented out code -- can it be removed?

Reply:
removed